### PR TITLE
Added attendee info and sales window icons

### PR DESCRIPTION
**Pull request by estebanhdzc**
### 🎫 Ticket

[ET-1824] 

### 🗒️ Description

- Added notes icon that on hover will display a tooltip with attendee fields
- Added sales window icon that on hover will display the sales dates

### 🎥 Artifacts 
image

image

image

image

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [ ] My code is tested. 
- [ ] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

### CODE REVIEW

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	const isTicketFuture = selectors.isTicketFuture( state, ownProps );
+	const isTicketPast = selectors.isTicketPast( state, ownProps );
+	const isTicketOnSale = selectors.isTicketOnSale( state, ownProps );
+	const saleWindowLabel = getSaleWindowLabel(isTicketFuture, isTicketPast, isTicketOnSale);
+
+	const startDateMoment = selectors.getTicketTempStartDateMoment( state, ownProps );
+	const endDateMoment = selectors.getTicketTempEndDateMoment( state, ownProps );
+	const fromDate = startDateMoment && startDateMoment.format('MMMM D, YYYY');
+	const toDate = endDateMoment && endDateMoment.format('MMMM D, YYYY');
```
Comment (bordoni) [inline src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:56]: Should respect the format we have saved on admin. Check with Juan which makes more sense here.

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+	return {
+		attendeeInfoFieldsLabel,
+		hasAttendeeInfoFields,
+		isBlockSelected: select('core/editor').getSelectedBlock() !== null,
```
Comment (bordoni) [inline src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:61]:
```suggestion
		isBlockSelected: select( 'core/editor' ).getSelectedBlock() !== null,
```

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+const mapStateToProps = ( state, ownProps ) => {
+	const attendeeInfoFields = selectors.getTicketAttendeeInfoFields( state, ownProps );
+	const attendeeInfoFieldsLabel = getAttendeeInfoFieldsLabel( attendeeInfoFields );
+	const hasAttendeeInfoFields = selectors.getTicketHasAttendeeInfoFields( state, ownProps );
+
+	const isTicketFuture = selectors.isTicketFuture( state, ownProps );
+	const isTicketPast = selectors.isTicketPast( state, ownProps );
+	const isTicketOnSale = selectors.isTicketOnSale( state, ownProps );
+	const saleWindowLabel = getSaleWindowLabel(isTicketFuture, isTicketPast, isTicketOnSale);
```
Comment (bordoni) [inline src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:51]:
```suggestion
	const saleWindowLabel = getSaleWindowLabel( isTicketFuture, isTicketPast, isTicketOnSale );
```

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+const getSaleWindowLabel = ( isTicketFuture, isTicketPast, isTicketOnSale ) => {
+	switch(true) {
+		case isTicketFuture:
+			return 'Scheduled';
+		case isTicketPast:
+			return 'Expired';
+		case isTicketOnSale:
+			return 'On sale';
+	}
```
Comment (bordoni) [inline src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:24]: These need to be translated, also use a map, dont pass all 3. Pass a `slug` makes a map easier.

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * Internal dependencies
+ */
+import Template from './template';
+import { withStore } from '@moderntribe/common/hoc';
+import { selectors } from '@moderntribe/tickets/data/blocks/ticket';
+
+const { select } = wp.data;
+
+const getSaleWindowLabel = ( isTicketFuture, isTicketPast, isTicketOnSale ) => {
```
Comment (bordoni) [inline src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:16]: Docblock.

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+			return 'Scheduled';
+		case isTicketPast:
+			return 'Expired';
+		case isTicketOnSale:
+			return 'On sale';
+	}
+	return '';
+}
+
+const getAttendeeInfoFieldsLabel = ( attendeeInfoFields ) => {
```
Comment (bordoni) [inline src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:28]: Docblock.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: readme.txt [hunk 7]
```diff
@@ -195,6 +195,7 @@ Check out our extensive [knowledgebase](https://evnt.is/18wm) for articles on us
 * Tweak - Re-styled Inactive RSVP Block in Block Editor to match frontend design. Styles will be the same in block editor and in the user interface [ET-1818]
 * Tweak - Re-styled Inactive Tickets Block in Block Editor. Empty state now has a new design [ET-1817]
 * Tweak - Re-styled Inactive Tickets Block with tickets. Inactive state with tickets has a new design [ET-1822]
+* Tweak - Re-styled Active Tickets Block with tickets. A notes icon will appear and on hover will give information about attendee information. A sales window icon will appear and on hover will give the sales dates information  [ET-1824]
```
Comment (codingmusician) [inline readme.txt:198]: Might be a little too wordy.
```suggestion
* Tweak - Re-styled Active Tickets Block with tickets. Add information icons and tooltips.  [ET-1824]
```

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/template.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+						propertyName={ __( 'Attendee registration', 'event-tickets' ) }
+						description={ attendeeInfoFieldsLabel }
+						icon={ <ARFIcon /> }
+					/>
+				)
+				: null
+			}
+
+			<IconWithTooltip
+				propertyName={ __( saleWindowLabel, 'event-tickets' ) }
```
Comment (codingmusician) [inline src/modules/blocks/ticket/container-header/title/attendee-registration-icons/template.js:45]: I would think `saleWindowLabel` should be translated before it gets passed into `AttendeeRegistrationIcons`.
```suggestion
				propertyName={ saleWindowLabel }
```

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js (hunk 1 above)
Comment (estebanhdzc) [reply src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js]: Thanks @bordoni @juanfra for your help. Its using the date format settings from wordpress now

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js (hunk 3 above)
Comment (estebanhdzc) [reply src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js]: Fixed

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js (hunk 5 above)
Comment (estebanhdzc) [reply src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:34]: Docblock was added

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js (hunk 4 above)
Comment (estebanhdzc) [reply src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js]: Thanks @bordoni, I implemented your suggestion

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js (hunk 6 above)
Comment (estebanhdzc) [reply src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:51]: The docblock was added

File: readme.txt (hunk 7 above)
Comment (estebanhdzc) [reply readme.txt]: Thanks @codingmusician I have updated the readme.txt as you suggested

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/template.js (hunk 8 above)
Comment (estebanhdzc) [reply src/modules/blocks/ticket/container-header/title/attendee-registration-icons/template.js]: `saleWindowLabel` is being recieved as translated now

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+	if ( attendeeInfoFields.length === 0 ) {
+		return '';
+	}
+
+	return attendeeInfoFields
+		.slice( 0, 4 )
+		.map( ( attendeeInformationField ) => attendeeInformationField.label )
+		.join( ', ' );
+}
```
Comment (bordoni) [inline src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:75]:
```suggestion
};
```

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+			return 'onSale';
+	}
+	return '';
+}
+
+/**
+ *
+ * @param { Array } attendeeInfoFields
+ * @returns { string } Returns the first 4 attendee fields joined by comma
+ */
```
[the remark this answers is not in the document]
Comment (estebanhdzc) [reply src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js]: Fixed

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js (hunk 9 above)
Comment (estebanhdzc) [reply src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js]: Fixed

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+ * @param { string } saleStatus
+ * @returns { string } The translated sale window label
+ */
+const getSaleWindowLabel = ( saleStatus ) => {
+	const labelMap = {
+		future: __( 'Scheduled', 'event-tickets' ),
+		past: __( 'Expired', 'event-tickets' ),
+		onSale: __( 'On sale', 'event-tickets' ),
+	};
+	return labelMap[ saleStatus ];
```
Comment (bordoni) [inline src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:33]: We need a conditional here in case the `saleStatus` doesn't exist.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js (hunk 11 above)
Comment (estebanhdzc) [reply src/modules/blocks/ticket/container-header/title/attendee-registration-icons/container.js:37]: Thanks. Validation was added
